tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -150,20 +188,40 @@ func (a *AwsProvider) CreateWindowsVM() (windowsVM types.WindowsVM, err error) {
 		return nil, fmt.Errorf("failed to get cluster worker IAM, %v", err)
 	}
 
-	// PowerShell script to setup WinRM for Ansible, installing OpenSSH server and open firewall
-	// port number 10250 on the Windows node created
-	userDataWinrm := `<powershell>
-        $url = "https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1"
-        $file = "$env:temp\ConfigureRemotingForAnsible.ps1"
-        (New-Object -TypeName System.Net.WebClient).DownloadFile($url,  $file)
-        & $file
-        Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
-        New-NetFirewallRule -DisplayName "` + types.FirewallRuleName + `"
-        -Direction Inbound -Action Allow -Protocol TCP -LocalPort ` + types.ContainerLogsPort + ` -EdgeTraversalPolicy Allow
+	publicKey, err := a.generatePublicKey()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate public key using private key: %v", err)
+	}
+
+	// PowerShell script to setup WinRM for Ansible, installing OpenSSH server, configuring
+	// public key authentication and open firewall port number 10250 on the Windows node created
+	userData := `<powershell>
+	$url = "https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1"
+	$file = "$env:temp\ConfigureRemotingForAnsible.ps1"
+	(New-Object -TypeName System.Net.WebClient).DownloadFile($url,  $file)
+	& $file
+	Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
+	New-NetFirewallRule -DisplayName "` + types.FirewallRuleName + `"
+	-Direction Inbound -Action Allow -Protocol TCP -LocalPort ` + types.ContainerLogsPort + ` -EdgeTraversalPolicy Allow
+	Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
+	Install-Module -Force OpenSSHUtils -Scope AllUsers
+	Set-Service -Name ssh-agent -StartupType ‘Automatic’
+	Set-Service -Name sshd -StartupType ‘Automatic’
+	Start-Service ssh-agent
+	Start-Service sshd
+	$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
+	$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+ 	$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
+	$passwordConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+	$pubKeyLocationConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'Match Group administrators','#Match Group administrators'
+	$pubKeyLocationConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+	Restart-Service sshd
+	New-item -Path $env:USERPROFILE -Name .ssh -ItemType Directory -force
+	echo "` + publicKey + `"| Out-File $env:USERPROFILE\.ssh\authorized_keys -Encoding ascii
```

Comment:
Please align

Reply:
The alignment I was asking for is similar to what was there before. Please do that when moving this to WMCO.